Algorithm/BinarySearch/leetcode/20. Valid Parentheses/claude sonnet 5/Valid_Parentheses_Java.md

```
@@ -0,0 +1,278 @@
+> ```
+
+> 💡 **ボクシング／アンボクシングを理解する**（初学者向け）
+>
+> **基礎**：Javaのプリミティブ型（`char`, `int`など）は、コレクション（`Deque`など）に直接格納できません。コレクションはオブジェクトしか扱えないため、`char`を`Character`という「箱」に入れる（ボクシング）必要があります。
+> ```java
+> Character boxed = 'a'; // 自動的にボクシングされる
+> char primitive = boxed; // 自動的にアンボクシングされる
+> ```
+>
+> **応用**：注意が必要なのは、`Character`同士を`==`で比較する場合です。Javaは`-128`〜`127`の範囲の値をキャッシュして使い回すため、この範囲内では`==`がたまたま`true`になりますが、範囲外では異なるオブジェクトとして扱われ`false`になることがあります。
+> ```java
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)